vllm/model_executor/layers/fused_moe/pplx_prepare_finalize.py

```diff
@@ -58,13 +60,22 @@ def prepare(
                 "apply_router_weight_on_input is only implemented for topk=1")
             a1 = a1 * rank_topk_weights.to(a1.dtype)
 
-        per_act_token = a1_scale.numel() != 1 if a1_scale is not None else (
-            a2_scale.numel() != 1 if a2_scale is not None else False)
-
-        a1q, a1q_scale = moe_kernel_quantize_input(a1, a1_scale,
-                                                   self.quant_dtype,
-                                                   per_act_token,
-                                                   self.block_shape)
+        repeat_cols = 4
+        if self.per_act_token:
+            repeat_rows = 1
+            a1q, a1q_scale = moe_kernel_quantize_input(a1, None,
+                                                       self.quant_dtype,
+                                                       self.per_act_token,
+                                                       self.block_shape)
+        else:
+            repeat_rows = a1.shape[0]
+            a1q, a1q_scale = moe_kernel_quantize_input(a1, a1_scale,
+                                                       self.quant_dtype,
+                                                       self.per_act_token,
+                                                       self.block_shape)
+
```

Comment:
nit: can you collapse these branches together?  e.g.
```
repeat_rows = 1 if self.per_act_token else a1.shape[0]
a1q, a1q_scale = moe_kernel_quantize_input(
                a1,
                a1_scale if not self.per_act_token else None,
                self.quant_dtype,
                self.per_act_token,
                self.block_shape)
```
or even simpler if `a1_scale` is None iff `self.per_act_token` you can just pass `a1_scale` directly.